rm/rm-node/src/main/java/org/ow2/proactive/resourcemanager/utils/KeyPairProducer.java

```diff
@@ -0,0 +1,136 @@
+    private static final TimeUnit timeUnit = TimeUnit.MILLISECONDS;
+
+    private AtomicInteger numberOfWorkers;
+
+    private Future<?> future;
+
+    private BlockingQueue<KeyPair> keyPairs = new LinkedBlockingQueue<>();
+
+    private long loopTimeout = 1000;
+
+    public KeyPairProducer() {
+        future = null;
+    }
+
+    public KeyPairProducer(int numberOfworkers) {
+        this.numberOfWorkers = new AtomicInteger(numberOfworkers);
+        final ExecutorService executorService = Executors.newSingleThreadExecutor();
+        future = executorService.submit(new Runnable() {
+            @Override
+            public void run() {
+                while (!Thread.interrupted()) {
+                    try {
+                        // if we obeserve less than 'numberOfWorkers' elements than we going to
+                        // add 'numberOfWorkers' more
+                        int numberOfKeysToAdd = numberOfWorkers.get() - keyPairs.size();
+                        if (numberOfKeysToAdd > 0) {
+                            for (int i = 0; i < numberOfKeysToAdd; ++i) {
+                                keyPairs.put(generateAndAddKeyPairs());
+                            }
+                        }
+
+                        // sleep 1 second, ot until notification (from getKeyPair method)
+                        synchronized (keyPairs) {
+                            keyPairs.wait(loopTimeout);
+                        }
+                    } catch (InterruptedException e) {
+                        LOGGER.error("Thread was interrupted.", e);
+                        Thread.currentThread().interrupt();
+                    } catch (Throwable e) {
+                        // ignore all exception except InterruptedException
+                        LOGGER.error(e);
+                    }
+                }
+            }
+        });
+    }
+
+    /**
+     * @return keyPair from head of the keyPairs queue
+     */
+    public KeyPair getKeyPair() throws NoSuchAlgorithmException {
+        // if by change queue is empty then we notify background thread which apparently sleeps
+        // so that we do not wait whole timeout.
+        if (keyPairs.isEmpty()) {
+            synchronized (keyPairs) {
+                keyPairs.notify();
+            }
+        }
+        try {
+            // block until there is something in the queue but do not block more than 3 timeouts
+            KeyPair keyPair = null;
+            if (future != null && !future.isDone()) {
+                keyPair = keyPairs.poll(loopTimeout * 3, timeUnit);
+                if (keyPair == null) {
+                    LOGGER.warn("Background thread is still working but there is nothing in the queue, so we generate key synchronously.");
+                    keyPair = generateAndAddKeyPairs();
+                }
+            } else {
+                //LOGGER.error("Background thread is dead, so we generate key synchronously.");
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') This block of commented-out lines of code should be removed. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3ACommentedOutCodeLine)